### ISSUE TITLE
Error 500 "Invalid mime type"

### ISSUE THREAD

**Issue by akirasan** [photo/openphoto-python#15]
Hello,

when I try to upload a photo by command line I have this error from the server:

{
    "code":500,
    "message":"Invalid mime type",
    "result":false
}

I exec this command line:

openphoto -p -X POST -H akirasan.trovebox.com  -e /photo/upload.json -F 'photo=/mnt/ksi2t/scripts/DSC_8422.JPG' -F 'tags=test'

The file is a jpeg image. Others request to the server trovebox.com are succesfull.

Regards,
akirasan

**Comment by jmathai** [comment]
I believe a local file needs to be prefixed with a @.

openphoto -p -X POST -H akirasan.trovebox.com -e /photo/upload.json -F 'photo=@/mnt/ksi2t/scripts/DSC_8422.JPG' -F 'tags=test'

**Comment by akirasan** [comment]
Yes, I try with the @ but the error is the same. I'm testing differents parameters like "--encode" at the end of the command line, but not work,...

**Comment by akirasan** [comment]
No code its implemented for upload image,.... :(

I've coding part of the solution and I upload successfully to my trovebox.

**Comment by sneakypete81** [comment]
One option is to Base64 encode the picture data. Something like:

import base64
encoded_photo = base64.b64encode(open(photo_file, "rb").read())

This is implemented in the https://github.com/photo/openphoto-python/pull/13 PR.

**Comment by walkah** [comment]
let me do another look through PR13 and we'll get that in and closed and hopefully clear this one up in the process! thanks @sneakypete81

**Comment by jmathai** [comment]
@akirasan If you have code to do the multipart upload then submit that in a separate pull request.

Thanks @akirasan @sneakypete81 @walkah

**Comment by akirasan** [comment]
Hi, here you are (I can't open a pull request):

---

```
def post(self, endpoint, params={}):
    url = urlparse.urlunparse(('http', self.host, endpoint, '', '', ''))

    if self.consumer_key:
        consumer = oauth.Consumer(self.consumer_key, self.consumer_secret)
        token = oauth.Token(self.token, self.token_secret)

        # ensure utf-8 encoding for all values.
        params = dict([(k, v.encode('utf-8')
                        if type(v) is types.UnicodeType else v)
                       for (k, v) in params.items()])

        files = [("photo", 'test.jpg', open('/mnt/ksi2t/scripts/test.jpg', 'rb').read())]

        import mimetools
        import mimetypes

        BOUNDARY = mimetools.choose_boundary()
        CRLF = '\r\n'
        L = []
        for (key, value) in params.items():
            L.append('--' + BOUNDARY)
            L.append('Content-Disposition: form-data; name="%s"' % key)
            L.append('')
            L.append(value)
        for (key, filename, value) in files:
            L.append('--' + BOUNDARY)
            L.append('Content-Disposition: form-data; name="%s"; filename="%s"' % (key, filename))
            L.append('Content-Type: %s' % mimetypes.guess_type(filename)[0] or 'application/octet-stream')
            L.append('')
            L.append(value)
            L.append('--' + BOUNDARY + '--')
            L.append('')
            body = CRLF.join(L)

        content_type = 'multipart/form-data; boundary=%s' % BOUNDARY

        headers = {'Content-Type': content_type, 'Content-Length': str(len(body))}
        client = oauth.Client(consumer, token)
        headers, content = client.request(url, "POST", body, headers)
        return content
```

---

Obviously my level in Python is very low, for this reason I hardcode the "file" path on: 
files = [("photo", 'test.jpg', open('/mnt/ksi2t/scripts/test.jpg', 'rb').read())]

...and when we call the command is not needed inform the path (this code is for only test :) ):
openphoto -p -X POST -H akirasan.trovebox.com -e /photo/upload.json

**Comment by sneakypete81** [comment]
Closed by #19 - thanks for your help @akirasan.

**Comment by akirasan** [comment]
Wow,...that's work perfectly!!!!,...great job!!!.
